src/main/java/org/terasology/structureTemplates/internal/ui/StructureTemplateEditorScreen.java

```diff
@@ -189,17 +195,13 @@ private void onMakeBoxShapedButton(UIWidget button) {
         });
     }
 
+    /**
+     * Compute the bounding box over the given regions, i.e., build the union of the given regions.
+     *
+     * @return the union of all regions; an invalid region of {@code regions} is empty
+     */
     private BlockRegion getBoundingRegion(List<BlockRegion> regions) {
-        if (regions.size() == 0) {
-            return new BlockRegion();
-        }
-        Vector3i min = new Vector3i(regions.get(0).getMin(new org.joml.Vector3i()));
-        Vector3i max = new Vector3i(regions.get(0).getMax(new org.joml.Vector3i()));
-        for (BlockRegion region: regions) {
-            min.min(region.getMin(new org.joml.Vector3i()));
-            max.max(region.getMax(new org.joml.Vector3i()));
-        }
-        return new BlockRegion(min, max);
+        return regions.stream().reduce(BlockRegions.INVALID, BlockRegion::union, BlockRegion::union);
```

Comment:
The semantics of this operation is "build the union over all the regions", but unfortunately we cannot easily overload `BlockRegions.encompassing(regions)` such that it works for all types which we can unify a region with.

Nevertheless, I think that this shorter code is less error-prone than doing computations on min/max of a block region manually.

Discord: https://discord.com/channels/270264625419911192/701818347238588561/790546593911799808